content/content.go

```diff
@@ -75,34 +77,76 @@ type RuleContentDirectory struct {
 	Rules  map[string]RuleContent
 }
 
+type fileContent struct {
+	content []byte
+	found   bool
+}
+
 // readFilesIntoByteArrayPointers reads the contents of the specified files
 // in the base directory and saves them via the specified byte slice pointers.
-func readFilesIntoByteArray(baseDir string, filelist []string) (map[string][]byte, error) {
-	var filesContent = map[string][]byte{}
+func readFilesIntoFileContent(baseDir string, filelist []string) (map[string]fileContent, error) {
+	var filesContent = map[string]fileContent{}
 	for _, name := range filelist {
+		log.Info().Msgf("Parsing %s/%s", baseDir, name)
 		var err error
-		ptr, err := ioutil.ReadFile(filepath.Clean(path.Join(baseDir, name)))
+		content := fileContent{}
+		rawBytes, err := ioutil.ReadFile(filepath.Clean(path.Join(baseDir, name)))
 		if err != nil {
-			return nil, err
+			content.content = nil
+			content.found = false
+			log.Error().Err(err)
+		} else {
+			content.content = rawBytes
+			content.found = true
 		}
-		filesContent[name] = ptr
+
+		filesContent[name] = content
 	}
+
 	return filesContent, nil
 }
 
 // readFilesIntoStringPointers reads the content of the specified files
 // in the base directory and saves them via the specified string pointer
-func readFilesIntoString(baseDir string, filelist []string) (map[string]string, error) {
-	var filesContent = map[string]string{}
-	for _, name := range filelist {
-		var err error
-		rawBytes, err := ioutil.ReadFile(filepath.Clean(path.Join(baseDir, name)))
-		filesContent[name] = string(rawBytes)
-		if err != nil {
-			return nil, err
-		}
+// func readFilesIntoString(baseDir string, filelist []string) (map[string]fileContent, error) {
```

Comment:
it's better to not have commented out code

Reply:
Done